cmd/query/app/flags.go

```diff
@@ -70,7 +70,9 @@ type QueryOptionsStaticAssets struct {
 // QueryOptionsBase holds configuration for query service shared with jaeger(v2)
 type QueryOptionsBase struct {
 	// BasePath is the base path for all HTTP routes
-	BasePath string
+	BasePath   string
+	// UIBasePath is the base path for Jaeger UI
```

Comment:
This comment does not clarify anything, I can infer from the variable name that is a "path for UI". I recommend adding a readme with a diagram (using mermaid markup, or perhaps just a table) that explains the difference. Participants in the diagram are browser, a reverse proxy, and query-service, and at each node show what the URL looks like and which config option is supposed to match that.